### PR TITLE
skip the target_clusters dupe ansible job check if commitID has been changed

### PR DESCRIPTION
**Pull request by mikeshng**
For https://github.com/open-cluster-management/backlog/issues/6324

It seems like new ansiblejobs are no longer fired if the commit has been changed (but `target_clusters` remain the same)
I think ansiblejobs should always fired if there is a commit and we detected it.

Signed-off-by: Mike Ng <ming@redhat.com>

### CODE REVIEW

File: pkg/controller/mcmhub/ansiblejob.go [hunk 1]
```diff
@@ -78,14 +79,14 @@ func (jIns *JobInstances) registryJobs(gClt GitOps, subIns *subv1.Subscription,
...
-		if forceRegister && len(jobRecords.Instance) != 0 {
+		if placementDecisionUpdated && len(jobRecords.Instance) != 0 {
 			plrSuffixFunc := func() string {
 				return fmt.Sprintf("-%v-%v", subIns.GetGeneration(), placementRuleRv)
 			}
 
 			suffix = plrSuffixFunc()
 
-			logger.Info("forceRegister suffix is: " + suffix)
+			logger.V(3).Info("placementDecisionUpdated suffix is: " + suffix)
```
Comment (ianzhang366) [inline pkg/controller/mcmhub/ansiblejob.go:89]: I dont't think `V(3)` will work for us, if you want it to be noise then `V(InfoLog)`, less noise `V(DebugLog)`

**Review by xiangjingli — APPROVED** [review]

File: pkg/controller/mcmhub/ansiblejob.go (hunk 1 above)
Comment (mikeshng) [reply pkg/controller/mcmhub/ansiblejob.go]: Using `DebugLog` now.

**Review by ianzhang366 — APPROVED** [review]

File: pkg/controller/mcmhub/ansiblejob.go [hunk 2]
```diff
@@ -95,7 +96,7 @@ func (jIns *JobInstances) registryJobs(gClt GitOps, subIns *subv1.Subscription,
 		logger.Info(fmt.Sprintf("registered ansiblejob %s", nxKey))
 
 		if _, ok := jobRecords.InstanceSet[nxKey]; !ok {
-			if forceRegister && len(jobRecords.Instance) > 0 {
+			if !commitIDChanged && placementDecisionUpdated && len(jobRecords.Instance) > 0 {
```
Comment (rokej) [inline pkg/controller/mcmhub/ansiblejob.go:99]: Can you add some logs here so when this is not run, we know from the logs?

Comment (mikeshng) [reply pkg/controller/mcmhub/ansiblejob.go]: added some more log please see new commit.

**Review by rokej — APPROVED** [review]
Thanks Mike

**Comment by mikeshng** [comment]
Undo part of the change in https://github.com/open-cluster-management/multicloud-operators-subscription/pull/328
that related to managed cluster resource status check.

**Review by ianzhang366 — APPROVED** [review]

**Pull request merged**